File: java/marker/src/main/java/com/github/wmacevoy/marker/BadMarkerState.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.github.wmacevoy.marker;

/**
 *
 * @author wmacevoy
 */
public class BadMarkerState extends Exception {
    private String problem;
    BadMarkerState(String problem) {
        this.problem = problem;
    }

    public String getProblem() {
        return problem;
    }

    public void setProblem(String problem) {
        this.problem = problem;
    }  
}

File: cpp/src/BadMarkerState.cpp
#include "BadMarkerState.h"

namespace marker {
    // std::string problem = "problem on line ";
    // problem.append(lineno);
    // BadMakerState bad(problem);

        BadMarkerState::BadMarkerState(const std::string &_problem) 
          : problem(_problem)
        {
        } 

        std::string BadMarkerState::getProblem() const {
            return problem;
        }
    
        void BadMarkerState::setProblem(const std::string &_problem) {
            problem = _problem;
        }

} // namespace marker